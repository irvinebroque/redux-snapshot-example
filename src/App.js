import React, { Component } from 'react';
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import ConnectedComponent from './ConnectedComponent'

function counter(state, action) {
  if (typeof state === 'undefined') {
    return 0
  }
  switch (action.type) {
    case 'INCREMENT':
      return state + 1
    case 'DECREMENT':
      return state - 1
    default:
      return state
  }
}

const store = createStore(counter)

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <div className="App-header">
            <h2>Welcome to React</h2>
          </div>
          <ConnectedComponent/>
        </div>
      </Provider>
    );
  }
}

export default App;
