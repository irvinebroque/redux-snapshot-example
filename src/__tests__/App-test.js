/* @flow */
import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import App from '../App'

test('App renders correctly', () => {
  const tree = renderer.create(<App/>).toJSON()
  expect(tree).toMatchSnapshot()
})
