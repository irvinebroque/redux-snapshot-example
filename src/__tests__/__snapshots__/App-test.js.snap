exports[`test App renders correctly 1`] = `
<div
  className="App">
  <div
    className="App-header">
    <h2>
      Welcome to React
    </h2>
  </div>
  <div>
    <div>
      some text
    </div>
    <div>
      <div>
        some text
      </div>
    </div>
  </div>
</div>
`;
